cert/selfsigned_test.go

```diff
@@ -185,3 +224,63 @@ func TestTLSDisableAutofill(t *testing.T) {
 		"TLS Certificate was not marked as outdated when it should be",
 	)
 }
+
+// TestTlsConfig tests to ensure we can generate a TLS Config from
+// a tls cert and tls key.
+func TestTlsConfig(t *testing.T) {
+	tempDir := t.TempDir()
+	certPath := tempDir + "/tls.cert"
+	keyPath := tempDir + "/tls.key"
+
+	// Generate TLS files with an extra IP and domain.
+	certBytes, keyBytes, err := cert.GenCertPair(
+		"lnd autogenerated cert", certPath, keyPath,
+		[]string{extraIPs[0]}, []string{extraDomains[0]}, false,
+		testTLSCertDuration,
+	)
+	require.NoError(t, err)
+
+	err = cert.WriteCertPair(certPath, keyPath, certBytes, keyBytes)
+	require.NoError(t, err)
+
+	certBytes, err = ioutil.ReadFile(certPath)
+	require.NoError(t, err)
+
+	keyBytes, err = ioutil.ReadFile(keyPath)
+	require.NoError(t, err)
+
+	// Load the certificate.
+	certData, parsedCert, err := cert.LoadCertFromBytes(
+		certBytes, keyBytes,
+	)
+	require.NoError(t, err)
+
+	// Check to make sure the IP and domain are in the cert.
+	var foundDomain bool
+	var foundIp bool
+	for _, domain := range parsedCert.DNSNames {
+		if domain == extraDomains[0] {
+			foundDomain = true
+			break
+		}
+	}
+	for _, ip := range parsedCert.IPAddresses {
+		if ip.String() == extraIPs[0] {
+			foundIp = true
+			break
+		}
+	}
+	if !foundDomain || !foundIp {
+		t.Fatal(fmt.Errorf("Did not find required information inside "+
+			"of TLS Certificate. foundDomain: %v, foundIp: %v",
+			foundDomain, foundIp))
+	}
+
+	// Create TLS Config.
+	tlsCfg := cert.TLSConfFromCert(certData)
+
+	assert.Equal(t, len(tlsCfg.Certificates), 1, fmt.Errorf("Found "+
```

Comment:
Use the `require` library here please. And no need to format an error message, the require library already creates a pretty nice error. So could just be `require.Len(t, tlsCfg.Certificates, 1)`.